src/aws-cpp-sdk-core/source/utils/crypto/factory/Factories.cpp

```diff
@@ -398,7 +402,9 @@ class DefaultAES_GCMFactory : public SymmetricCipherFactory
         Aws::Crt::Optional<Aws::Crt::ByteCursor> tagCur = tag.GetLength() > 0 ? Aws::Crt::ByteCursorFromArray(tag.GetUnderlyingData(), tag.GetLength()) : Aws::Crt::Optional<Aws::Crt::ByteCursor>();
         Aws::Crt::Optional<Aws::Crt::ByteCursor> aadCur = aad.GetLength() > 0 ? Aws::Crt::ByteCursorFromArray(aad.GetUnderlyingData(), aad.GetLength()) : Aws::Crt::Optional<Aws::Crt::ByteCursor>();
 
-        return Aws::MakeShared<CRTSymmetricCipher>(s_allocationTag, Aws::Crt::Crypto::SymmetricCipher::CreateAES_256_GCM_Cipher(keyCur, ivCur, tagCur, aadCur));
+        auto cipher = Aws::Crt::Crypto::SymmetricCipher::CreateAES_256_GCM_Cipher(keyCur, ivCur, aadCur);
+        cipher.SetTag(tagCur.has_value() ? tagCur.value() : Aws::Crt::ByteCursor{});
```

Comment:
nit: pull into if and only set if tagCur has value.
does matter much and should still work as is.

ideally tag is only set for decrypt operations, but with the way cpp sdk interface is now, its probably not a possible change